services/operator_info.go

```diff
@@ -0,0 +1,85 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Rocket-Rescue-Node/credentials"
+	"github.com/Rocket-Rescue-Node/rescue-api/models"
+	"go.uber.org/zap"
+)
+
+type OperatorInfo struct {
+	CredentialEvents []int64          `json:"credentialEvents"`
+	QuotaSettings    *json.RawMessage `json:"quotaSettings,omitempty"`
+}
+
+func (s *Service) GetOperatorInfo(msg []byte, sig []byte, ot credentials.OperatorType) (*OperatorInfo, error) {
+	var err error
+
+	// Check request age
+	if err := s.checkRequestAge(&msg); err != nil {
+		return nil, err
+	}
+
+	// Recover nodeID
+	nodeID, err := s.getNodeID(&msg, &sig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check node authz
+	if err := s.checkNodeAuthorization(nodeID, ot); err != nil {
+		return nil, err
+	}
```

Comment:
These three functions are called in the exact same sequence as the other service.

Can we add a function to service.go with:
```golang
func (s *Service) validateSignedRequest(msg []byte, sig []byte, ot credentials.OperatorType) (*common.Address, error)
```

whose body is defined by simply calling these three functions? Then deduplicate across both services.

Reply:
At some point I had it written that way, but I was worried it "hid" too much of the validation logic. Glad you made this suggestion. :) I'll make the change.